### PR TITLE
Refactor: Make all chat area use graphql

### PR DESCRIPTION
**Pull request by Tainan404**
### What does this PR do?

The purpose of this pull request is to revamp the chat panel completely by implementing GraphQL. Additionally, it involves discontinuing the subscription to the chat collection and chat messages, as well as ceasing the monitoring of Redis messages associated with them.

NOTE: At present, no code has been deleted. Instead, we have modified it by removing the export of certain components. This adjustment was made with the intention of simplifying future merges between versions and providing clarity for potential UI fixes.

### CODE REVIEW

**Comment by gustavotrott** [comment]
When I first open a Private chat and try to send a message, the message is not being sent!
If I switch chat and open the private again, then the send-message works!


https://github.com/bigbluebutton/bigbluebutton/assets/5660191/4d49bf0b-e013-444b-9232-f35f89e93898

**Comment by gustavotrott** [comment]
Sometimes the user is seeing the msg but is is marked as unread in the ChatList:

https://github.com/bigbluebutton/bigbluebutton/assets/5660191/46db1dcd-ea5b-4599-b7f8-8b8c52f2830f

**Comment by gustavotrott** [comment]
Sometimes the scroll is not on the bottom. Then the msgs is correctly not marked as read in the Userlist.
But then I scroll to the bottom, but it mgs keep maked as unread!

https://github.com/bigbluebutton/bigbluebutton/assets/5660191/bc6d581e-549f-4b7b-ac08-d3e23cb78b3f

File: bigbluebutton-html5/imports/ui/components/chat/chat-graphql/chat-message-form/component.tsx [hunk 1]
```diff
@@ -198,7 +212,7 @@ const ChatMessageForm: React.FC<ChatMessageFormProps> = ({
   const handleMessageChange = (e: React.FormEvent<HTMLInputElement>) => {
     let newMessage = null;
     let newError = null;
-
+    console.log('handleMessageChange', e)
```
Comment (gustavotrott) [inline bigbluebutton-html5/imports/ui/components/chat/chat-graphql/chat-message-form/component.tsx:215]:
```suggestion
```

File: bigbluebutton-html5/imports/ui/components/chat/chat-graphql/chat-message-list/page/chat-message/componet.tsx [hunk 2]
```diff
@@ -19,62 +17,107 @@ interface ChatMessageProps {
...
   },
 });
 
-const ChatMesssage: React.FC<ChatMessageProps> = ({ message, previousMessage, lastSenderPreviousPage}) => {
+const ChatMesssage: React.FC<ChatMessageProps> = ({ message, previousMessage, lastSenderPreviousPage }) => {
   const intl = useIntl();
   if (!message) return null;
   const sameSender = (previousMessage?.user?.userId || lastSenderPreviousPage) === message?.user?.userId;
   const dateTime = new Date(message?.createdTime);
+  console.log('message', message);
```
Comment (gustavotrott) [inline bigbluebutton-html5/imports/ui/components/chat/chat-graphql/chat-message-list/page/chat-message/componet.tsx:51]:
```suggestion
```

**Comment by gustavotrott** [comment]
The tail is not working correctly!
Take a look at the right side window, after 27 seconds the scroll starts to go up automatically, when it was supposed to be stuck at the bottom showing always the last msg received!

https://github.com/bigbluebutton/bigbluebutton/assets/5660191/2841608d-0765-499e-b640-9c34c484906c

**Comment by gustavotrott** [comment]
Change this button label to "Show Welcome Message" as suggested by AntonG.

image

**Comment by gustavotrott** [comment]
I think this button should be hidden when there are no Welcome Msgs to show!

image

**Comment by gustavotrott** [comment]
The rule that decide if it will group the messages of a given user is inconsistent!
Sometimes it doesn't group the messages, but then it reives a new message and it begins to group them again! And then receiving a new message, it separates the previously grouped messages into two different groups again.

Take a look at second 30 in the video.
All the messages from the "Teacher" are initially grouped together. However, when "Student 1" sends a message, the Teacher's messages become split into two separate groups. Then, once the Teacher sends another message, his messages revert back to being grouped together.

https://github.com/bigbluebutton/bigbluebutton/assets/5660191/2e5ace2c-2ddf-44f1-b434-8aab8846c795

- Not grouped:
not-grouped

- Then grouped again:
grouped

**Comment by gustavotrott** [comment]
It' not showing the button "Load more"
When I scroll to the top it automatically load all messages available!
It it on purpose?

https://github.com/bigbluebutton/bigbluebutton/assets/5660191/d19b9880-960b-4924-a28f-050c1848e2ea

File: bigbluebutton-html5/imports/ui/components/chat/chat-graphql/chat-message-form/component.tsx [hunk 3]
```diff
@@ -317,38 +334,69 @@ const ChatMessageForm: React.FC<ChatMessageFormProps> = ({
...
       idChatOpen,
       handleClickOutside,
-      chatAreaId,
-      chatId,
-      chatTitle,
-      connected,
-      disabled,
-      partnerIsLoggedOut,
+      chatId: idChatOpen,
+      connected: true, //TODO: monitoring network status
```
Comment (TiagoJacobs) [inline bigbluebutton-html5/imports/ui/components/chat/chat-graphql/chat-message-form/component.tsx:394]: Please open an issue referring this TODO - with the 2.8 milestone

**Comment by gustavotrott** [comment]
The private chat has some problems when informing that the message is Read!

https://github.com/bigbluebutton/bigbluebutton/assets/5660191/3a8a8cfc-324c-404e-8d64-5ba705523a5c

**Review by gustavotrott — APPROVED** [review]
LGTM!

**Pull request merged**